TutorialBot/TutorialBotOutput.py

```diff
@@ -0,0 +1,108 @@
+import math
+import tensorflow as tf
+
+
+def get_output_vector(values, given_output):
+    def distance(x1, y1, x2, y2):
+        return tf.sqrt(tf.square(x2 - x1) + tf.square(y2 - y1))
+
+    def to_degrees(radians):
+        return radians * 180 / math.pi
+
+    def aim(target_x, target_y):
+        angle_between_bot_and_target = math.degrees(math.atan2(target_y - bot_pos.Y,
+                                                               target_x - bot_pos.X))
+        to_degrees(tf.atan2(tf.subtract(target_y, bot_pos.Y), tf.subtract(target_x, bot_pos.X)))
+        angle_front_to_target = angle_between_bot_and_target - bot_yaw
+
+        # Correct the values
+        angle_front_to_target += tf.cond(tf.less(angle_front_to_target, -180), lambda: 360,
+                                         lambda: tf.cond(tf.greater(angle_front_to_target, 180), lambda: -360,
+                                                         lambda: 0))
+
+        st = tf.cond(tf.less(angle_front_to_target, -10), lambda: -1,
+                     lambda: tf.cond(tf.greater(angle_front_to_target, 10), lambda: 1, lambda: 0))
+
+        ps = tf.cond(tf.less(tf.abs(to_degrees(angle_front_to_target)), powerslide_angle), lambda: True,
+                     lambda: False)
+        return [st, ps]
+
+    index = 0
+
+    # Contants
+    distance_from_ball_to_boost = tf.constant(1500)  # Minimum distance to ball for using boost
+    powerslide_angle = tf.constant(170)  # The angle (from the front of the bot to the ball) to start to powerslide.
+
+    # Controller inputs
+    # throttle = 0 defined in 100
+    steer = 0
+    pitch = 0
+    yaw = 0
+    roll = 0
+    # boost = FalseThey
+    jump = False
+    powerslide = tf.constant(False)
+
+    # Game values
+    bot_pos = None
+    bot_rot = None
+    ball_pos = None
+    bot_yaw = None
+
+    # Update game data variables
+    bot_pos = values.gamecars[index].Location
+    bot_rot = values.gamecars[index].Rotation
+    ball_pos = values.gameball.Location
+
+    # Get car's yaw and convert from Unreal Rotator units to degrees
+    bot_yaw = tf.abs(bot_rot.Yaw) % 65536 / 65536 * 360
```

Comment:
this is a little unclear about the order in which it is executed maybe make that more obvious